frontend/server/src/Controllers/Contest.php

```diff
@@ -501,6 +508,66 @@ public static function getContestDetailsForSmarty(
         return $result;
     }
 
+    /**
+     * @return array{payload: array{status: string, contests: array{contest_id: int, problemset_id: int, acl_id?: int, title: string, description: string, original_finish_time?: string, start_time: int|null, finish_time: int|null, last_updated: int|null, window_length: null|int, rerun_id: int, admission_mode: string, alias: string, scoreboard?: int, points_decay_factor?: float, partial_score?: int, submissions_gap?: int, feedback?: string, penalty?: int, penalty_type?: string, penalty_calc_policy?: string, show_scoreboard_after?: int, urgent?: int, languages?: null|string, recommended: int, scoreboard_url: string, scoreboard_url_admin: string}[]}, privateContestsAlert: bool}
+     */
+    public static function getContestListMineForSmarty(
+        \OmegaUp\Request $r
+    ): array {
+        // Authenticate user
+        $r->ensureMainUserIdentity();
+
+        // If the user have private material (contests/problems), an alert is issued
+        // suggesting to contribute to the community by releasing the material to
+        // the public. This flag ensures that this alert is shown only once per
+        // session, the first time the user visits the "My contests" page.
+        $privateContestsAlert = (
+            !isset($_SESSION['private_contests_alert']) &&
+            \OmegaUp\DAO\Contests::getPrivateContestsCount($r->user) > 0
+        );
+
+        if ($privateContestsAlert) {
+            $_SESSION['private_contests_alert'] = true;
+        }
+
+        return [
+            'payload' => self::getContestListInternal(
+                $r,
+                function (
+                    int $identityId,
+                    int $page,
+                    int $pageSize,
+                    ?string $query
+                ) {
+                    return \OmegaUp\DAO\Contests::getAllContestsOwnedByUser(
+                        $identityId,
+                        $page,
+                        $pageSize
+                    );
+                }
+            ),
+            'privateContestsAlert' => $privateContestsAlert,
+        ];
+    }
+
+    /**
+     * @return array{LANGUAGES: string[], IS_UPDATE?: int}
+     */
+    public static function getContestNewDetailsForSmarty(
+        \OmegaUp\Request $r,
+        bool $isUpdate = false
+    ): array {
+        $result = [
+            'LANGUAGES' => array_keys(
+                \OmegaUp\Controllers\Run::SUPPORTED_LANGUAGES
+            ),
```

Comment:
será posible cambiar el template para que acepte un `IS_UPDATE` booleano en vez de entero? y podérselo pasar incondicionalmente:

```suggestion
        return [
            'LANGUAGES' => array_keys(
                \OmegaUp\Controllers\Run::SUPPORTED_LANGUAGES
            ),
            'IS_UPDATE' => $isUpdate,
        ];
```

Reply:
Listo.